src/main/java/racingcar/model/RaceGame.java

```diff
@@ -0,0 +1,52 @@
+package racingcar.model;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import racingcar.util.RaceCountValidator;
+import racingcar.view.OutputView;
+
+public class RaceGame {
+
+
+    private final Move move;
+
+    public RaceGame(Move move) {
+        this.move = move;
+
+    }
+
+    public String playGame(List<Car> racingCars, String  raceCountInput) {
+        int raceCount = parseRaceCount(raceCountInput);
+        List<Car> gameResult = printRaceResult(racingCars, raceCount);
+
+        return compareWinner(gameResult);
+
+    }
+
+    private static int parseRaceCount(String raceCount) {
+        RaceCountValidator.validate(raceCount);
+        return Integer.parseInt(raceCount);
+    }
+
+    private List<Car> printRaceResult(List<Car> cars, int raceCount) {
+
+        for (int i = 0; i < raceCount; i++) {
+            move.moveAccordingToRandomNumber(cars);
+            OutputView.printRaceProgress(cars);
```

Comment:
이동 결과를 출력하려면 OutputView가 필요한데 static 메서드라 OutputView를 주입받지 않아도 되군요 👍 👍 

하지만, 결과를 출력할 때 저도 OutputView 끌어다 썼는데, 승환님이 리뷰하신 부분이 View와 Service간 역할이 겹치는 부분을 분리 시켜서 Controller가 조율하도록 하라는 피드백이 생각이 났습니다. 이 방법은 저도 못 해냈기 때문에 잘 모르지만, 승환님 말씀이 맞는 것 같다고 생각합니다..

어떻게 생각하시나요? 
```suggestion
public class RaceController {

    public void startRace(List<Car> cars, int raceCount) {
        for (int i = 0; i < raceCount; i++) {
            raceService.moveCars(cars); // 비즈니스 로직을 수행
            OutputView.printRaceProgress(cars); // 결과 출력
        }
 }


```
이런 느낌으로 gpt 도움 받았습니다..  하지만 위에 부분도 controller에서 for문을 수행한다는 게 굉장히 어색하네요.. 어떤 방법이 있을지 회의해보면 좋을 것 같습니다~

Reply:
예제코드까지 주신 점 감사합니다! 어제 오프라인 스터디에서 이야기를 나누고 혼자 더 생각해봤는데 현재 코드는 MVC 패턴의 역할 분리 관점에서 위반되어 리팩터링하는 게 좋을 것 같습니다. 그래서 제가 생각한  두 가지 해결 방법이 있습니다.

1.controller에서 model의 값을 받아 view로 넘겨 출력한다.
-> 기존 `printRaceResult`의 return을 void로 바꾸고, 이 함수를 `RaceGameController`에서 실행한다.

2. 경주 과정을 저장하여 controller에 보내고 저장된 결과 view에서 출력한다.
```java
public String raceProgress(List<Car> cars, int raceCount) { 
    StringBuilder carsRaceProgress = new StringBuilder(); // StringBuilder 초기화

    for (int i = 0; i < raceCount; i++) {
        raceService.moveCars(cars); // 비즈니스 로직을 수행
        appendRaceResults(cars, carsRaceResults); // 결과를 StringBuilder에 추가
    }

    return carsRaceProgress.toString(); // 최종 결과 문자열 반환
}

private void appendRaceResults(List<Car> cars, StringBuilder carsRaceResults) {
    for (Car car : cars) {
        carsRaceResults.append(car.getCarName())
                       .append(" : ")
                       .append(car.getMoveCount())
                       .append("\n");
    }
    carsRaceResults.append("\n");
}
```

-> controller에서  raceProgress의 결과를 output view에 넘겨주어 출력하도록 한다.



저는 2번 방법이 더 좋은 것 같습니다.

`RaceGameController`는 `RaceGame`의 내부 로직이나 데이터 구조에 대해 구체적으로 알 필요 없이 오직 게임의 결과만을 `View`에 전달합니다.

구체적으로`RaceGameController`는 `playGame`의 결과로 단순히 `String`을 받아 출력할 뿐이며, 로직의 세부 구현이 감춰져 있기 때문에 게임 로직이 수정되거나 `RaceGame` 내부 데이터 타입이 변경되더라도 컨트롤러와 뷰에 영향을 미치지 않습니다. 따라서 확장 가능성도 높아져 2번 방법으로 리팩터링 하는 게 좋을 것 같습니다!